### PR TITLE
Lint for common test errors

### PR DESCRIPTION
**Pull request by Vinnl**
And fix the existing ones identified by the linter.

### CODE REVIEW

File: src/app/(proper_react)/redesign/(authenticated)/user/welcome/Onboarding.test.tsx [hunk 1]
```diff
@@ -112,9 +112,7 @@ it("form input elements have invalid state if left empty on step 2", async () =>
   expect(firstNameInput.getAttribute("aria-invalid")).toBe(null);
   expect(lastNameInput.getAttribute("aria-invalid")).toBe(null);
   expect(locationInput.getAttribute("aria-invalid")).toBe(null);
-  if (dateInput) {
-    expect(dateInput.getAttribute("aria-invalid")).toBe(null);
-  }
+  expect(dateInput?.getAttribute("aria-invalid")).toBe(null);
```
Comment (Vinnl) [inline src/app/(proper_react)/redesign/(authenticated)/user/welcome/Onboarding.test.tsx:115]: Conditional `expect`s wouldn't have failed the the `dateInput` inadvertently didn't exist.

File: src/app/components/client/toolbar/UserMenu.test.tsx [hunk 2]
```diff
@@ -38,20 +38,6 @@ it("opens and closes the user menu", async () => {
   expect(screen.queryByText("example@example.com")).not.toBeInTheDocument();
 });
 
-it("opens and closes the user menu", async () => {
```
Comment (Vinnl) [inline src/app/components/client/toolbar/UserMenu.test.tsx:41]: This test had the exact same name as the one preceding it, and as far as I can see, _is_ exactly the same.

File: src/app/functions/server/featureFlags.test.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ it("returns flag state if exists", async () => {
       };
     } else {
       console.debug("flag name:", name);
-      fail("Unknown flag passed to mock");
+      throw new Error("Unknown flag passed to mock");
```
Comment (Vinnl) [inline src/app/functions/server/featureFlags.test.ts:52]: This was apparently an invalid use of `fail`, and throwing an error was preferred according to the lint message.

File: src/app/components/client/toolbar/UserMenu.test.tsx (hunk 2 above)
Comment (flozia) [reply src/app/components/client/toolbar/UserMenu.test.tsx:41]: Oops — testing it twice does not make things better?

**Review by flozia — APPROVED** [review]
Great addition!

File: src/app/components/client/toolbar/UserMenu.test.tsx (hunk 2 above)
Comment (Vinnl) [reply src/app/components/client/toolbar/UserMenu.test.tsx:41]: 100% more testing is 100% more better!

**Pull request merged**